File: python/twosum.py
def two_sum(nums: list[int], target: int) -> list[int]:
    m = {}
    for i, n in enumerate(nums):
        if m.get(n) is not None:
            return [i, m[n]]
        m[target - n] = i
    return []


print(two_sum([11, 15, 2, 7], 9))
print(two_sum([3, 2, 3], 6))
print(two_sum([5, 6, 12, 22, 41, 5, 63, 14, 29, 0, 0, 1, 0], 104))

File: hashtables/two_sum.go
/*
1. Two Sum

Given an array of integers nums and an integer target, return indices of the two
numbers such that they add up to target.
You may assume that each input would have exactly one solution, and you may not use the same element twice.
You can return the answer in any order.
*/

package hashtables

func twoSum(nums []int, target int) []int {
	hm := map[int]int{nums[0]: 0}
	for i := 1; i < len(nums); i++ {
		diff := target - nums[i]
		if seen, ok := hm[diff]; ok {
			return []int{seen, i}
		}
		hm[nums[i]] = i
	}
	return []int{}
}
